tsinfer/inference.py

```diff
@@ -1907,6 +1909,149 @@ def get_mutations(self, node_id):
         return self.mutations[node_id]
 
 
+def has_single_edge_over_grand_root(ts):
+    # Internal function to check if this is a "raw" inferred tree sequence.
+    if ts.num_edges < 2:
+        # must have edge to grand root and above grand root
+        return False
+    last_edge = ts.edge(-1)
```

Comment:
I'd find it easier to follow the logic here (and it would be easier to test) if the full-span requirement was tested first. This clause will also catch split edges like (0, x, u, vr), (x, L, u, vr).